### PR TITLE
refactor(api): topic_metrics api swagger spec

### PR DESCRIPTION
**Pull request by JimMoen**

### CODE REVIEW

**Review by zhongwencool — APPROVED** [review]

File: apps/emqx_modules/src/emqx_topic_metrics_api.erl [hunk 1]
```diff
@@ -18,121 +18,219 @@
...
-
-topic_param() ->
-    #{
-        name => topic,
-        in => path,
-        required => true,
-        description => <<"Notice: Topic string url must encode">>,
-        schema => #{type => string}
+     , delete =>
+           #{ description => <<"Deregister topic metrics">>
```
Comment (HJianBo) [inline apps/emqx_modules/src/emqx_topic_metrics_api.erl:98]:
```suggestion
           #{ description => <<"Remove the topic metrics">>
```

File: apps/emqx_modules/src/emqx_topic_metrics_api.erl [hunk 2]
```diff
@@ -18,121 +18,219 @@
...
+            , tags => ?API_TAG_MQTT
+            , parameters => [topic(path)]
+            , responses =>
+                  #{ 204 => <<"Deregister topic metrics">>,
```
Comment (HJianBo) [inline apps/emqx_modules/src/emqx_topic_metrics_api.erl:102]:
```suggestion
                  #{ 204 => <<"Removed topic metrics">>,
```

File: apps/emqx_modules/src/emqx_topic_metrics_api.erl [hunk 3]
```diff
@@ -18,121 +18,219 @@
...
+                     404 => emqx_dashboard_swagger:error_codes([?ERROR_TOPIC], <<"Topic not found">>)
+                   }
+            }
+     }.
+
+fields(reset) ->
+    [ {topic
+      , mk( binary()
+          , #{ desc => <<"No topic will reset all">>
```
Comment (HJianBo) [inline apps/emqx_modules/src/emqx_topic_metrics_api.erl:111]:
```suggestion
          , #{ desc => <<"Topic Name. If this paramter is not present,all created  topic metrics will be reset">>
```

File: apps/emqx_modules/src/emqx_topic_metrics_api.erl [hunk 4]
```diff
@@ -18,121 +18,219 @@
...
+      { metrics
+      , mk( ref(metrics)
+          , #{ desc => <<"MQTT Topic Metrics">>
+             , nullable => false})
+      }
+    ];
+
+fields(metrics) ->
+    [ { 'messages.dropped.count'
+      , mk( integer(), #{ desc => <<"MQTT Message dropped count.">>
```
Comment (HJianBo) [inline apps/emqx_modules/src/emqx_topic_metrics_api.erl:145]: We should remove the `MQTT` qualifier. e.g, this message could also be published from MQTT-SN or CoAP

Comment (JimMoen) [reply apps/emqx_modules/src/emqx_topic_metrics_api.erl:145]: But How to describe the `Message with QoS 1 received count.`
Is there a concept of QoS in CoAP protocol?

**Review by HJianBo — APPROVED** [review]

**Review by HJianBo — APPROVED** [review]

**Pull request merged**